tests/pipeline/test_resources_evaluation.py

```diff
@@ -212,99 +215,174 @@ async def async_resource1():
-#             if inspect.isfunction(gen):
-#                 gen = gen()
-#             # if we have an async gen, no further action is needed
-#             if inspect.isasyncgen(gen):
-#                 raise Exception("Already async gen")
-
-#             # get next item from generator
-#             def _gen():
-#                 nonlocal exhausted
-#                 # await asyncio.sleep(0.1)
-#                 try:
-#                     return next(gen)
-#                 # on stop iteration mark as exhausted
-#                 except StopIteration:
-#                     exhausted = True
-#                     return None
-#                 finally:
-#                     nonlocal busy
-#                     busy = False
-
-#             try:
-#                 while not exhausted:
-#                     while busy:
-#                         yield None
-#                     busy = True
-#                     yield _gen
-#             except GeneratorExit:
-#                 # clean up inner generator
-#                 gen.close()
-
-#         return _wrap
-
-#     @parallelize
-#     def resource1():
-#         for l_ in ["a", "b", "c"]:
-#             time.sleep(0.5)
-#             nonlocal execution_order
-#             execution_order.append("one")
-#             threads.add(threading.get_ident())
-#             yield {"letter": l_}
-
-#     @parallelize
-#     def resource2():
-#         time.sleep(0.25)
-#         for l_ in ["e", "f", "g"]:
-#             time.sleep(0.5)
-#             nonlocal execution_order
-#             execution_order.append("two")
-#             threads.add(threading.get_ident())
-#             yield {"letter": l_}
-
-#     @dlt.source
-#     def source():
-#         if parallelized:
-#             return [resource1(), resource2()]
-#         else:  # return unwrapped resources
-#             return [resource1.__wrapped__(), resource2.__wrapped__()]
-
-#     pipeline_1 = dlt.pipeline("pipeline_1", destination="duckdb", full_refresh=True)
-#     pipeline_1.run(source())
-
-#     # all records should be here
-#     with pipeline_1.sql_client() as c:
-#         with c.execute_query("SELECT * FROM resource1") as cur:
-#             rows = list(cur.fetchall())
-#             assert len(rows) == 3
-#             assert {r[0] for r in rows} == {"a", "b", "c"}
-
-#         with c.execute_query("SELECT * FROM resource2") as cur:
-#             rows = list(cur.fetchall())
-#             assert len(rows) == 3
-#             assert {r[0] for r in rows} == {"e", "f", "g"}
-
-#     if parallelized:
-#         assert len(threads) > 1
-#         assert execution_order == ["one", "two", "one", "two", "one", "two"]
-#     else:
-#         assert execution_order == ["one", "one", "one", "two", "two", "two"]
-#         assert len(threads) == 1
+@pytest.mark.parametrize("parallelized", [True, False])
+def test_parallelized_resource(parallelized: bool) -> None:
+    os.environ["EXTRACT__NEXT_ITEM_MODE"] = "fifo"
```

Comment:
what for round robin? test it for both. I want to see what is the difference!

Reply:
Yeah can parametrize it. Should be not much difference. It's going round-robin implicitly but resetting at some points depending on timings.